### PR TITLE
docs: New fiddle example 'Window Crash'

### PR DESCRIPTION
**Pull request by ValerianClerc**
#### Description of Change
Added a fiddle example for window crash handling. Based off of code from [electron-api-demos](https://github.com/electron/electron-api-demos)

For: #20442 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Added "Relaunch window after process crashes" fiddle example to docs

### CODE REVIEW

File: docs/fiddles/windows/crashes-and-hangs/crash/index.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    bottom: 20px;
+    left: 50%;
+    transform: translateX(-50%);
+    font-size: 12px;
+    text-decoration: none;
+  }
+</style>
+
+</head><body><p>Click the text below to crash and then reload this process.</p>
+<a id="crash" href="javascript:process.crash()" >Crash this process</a>
```
Comment (felixrieseberg) [inline docs/fiddles/windows/crashes-and-hangs/crash/index.html]:
```suggestion
<a id="crash" href="javascript:process.crash()">Crash this process</a>
```

File: docs/fiddles/windows/crashes-and-hangs/crash/index.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
+<html><head><style>
+  body {
```
Comment (felixrieseberg) [inline docs/fiddles/windows/crashes-and-hangs/crash/index.html]: Can you do us a favor and cut out the `style`? Thank you!

**Comment by ValerianClerc** [comment]
> Thank you very much! We just need to cut the CSS, it's otherwise good to go.

Alright all done! @felixrieseberg 

Also I noticed that the `appveyor: win-ia32-testing-pr` build failed, do you know what that's about? It's saying that tests failed, but I didn't write any tests or change any tested code.

Thanks!

**Comment by erickzhao** [comment]
@ValerianClerc CI logs seem to show that the tests are failing: `Electron tests failed with code 0xc0000374.`

As this is a documentation-only PR and doesn't modify Electron internals, I wouldn't worry about it. Tests flake fairly often!

**Comment by zcbenz** [comment]
@ValerianClerc Do you still plan to work on this PR per @erickzhao's review?

**Comment by ValerianClerc** [comment]
Hi @zcbenz, thank you for the reminder, I'll keep working on this one and wrap it up.

@erickzhao Thanks for the comments and sorry for the delay, I promisified the dialog MessageBox to fit the newer APIs! (tested on 7.1.3)

I tried to do the crash recovery with `mainWindow.reload()`/`mainWindow.close()` initially, because that's what they do in the `electron-api-demos` app. But I get the same stack trace crash as you.

Could you point me in the direction to tackle this? My thoughts so far:
1. From my understanding, in `electron-api-demos`, the code is being executed as a child window to the main app, in the `renderer.js`. However, from your comments on the `createWindow` PR #20480, I understand that `electron-fiddle` doesn't support loading multiple HTML windows? So I can't make a child window with a button to crash itself, since I already have an HTML file loaded for the parent window? And when the parent window (app) crashes, I have to reload the entire app?
2. Could also just be that `electron-api-demos` is outdated

**Comment by zcbenz** [comment]
@ValerianClerc I'm jumping into the discussion.

The usual way of recovering crash is to close the window and create a new one, which is used by apps like Slack. Reloading a crashed window might not work.

I'm not sure why `mainWindow.close()` crashes, but we should use the right approach here and fix the crash instead. If the crash still happens can you create an issue for it?

> 1. From my understanding, in `electron-api-demos`, the code is being executed as a child window to the main app, in the `renderer.js`

The fiddle examples are executed as independent apps, so it is fine create multiple windows in this app.

**Comment by codebytere** [comment]
Closing as abandoned.

**Pull request closed without merging**